helpers/localizePrice.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+const common = require('./lib/common.js');
+
+const factory = globals => {
+    return function(price, locale) {
+
+        if (!common.isObject(price) || !common.isString(price.currency)
+            || isNaN(price.value) || !common.isString(price.formatted)) {
+            // Return empty string if this does not appear to be a price object
+            return ''
+        }
+
+        if (!common.isString(locale) || locale.length < 2) {
+            // Valid browser language strings are at least two characters
+            // https://www.metamodpro.com/browser-language-codes
+            // If provided locale is less than two characters (or not a string),
+            // return the normal formatted price
+            return price.formatted;
+        }
+
+        // Try to format the price to the provided locale,
+        // but if anything goes wrong,
+        // just return the usual price
+        try {
+            return new Intl.NumberFormat(
```

Comment:
Is the image of Node used by our storefront renderer built with ICU support? I think only a subset of the full ICU data is supported by default (https://nodejs.org/docs/latest-v10.x/api/intl.html). If it's not, we might need to build with a different image in order to support more than just English locale.

Not sure how to handle the `stencil-cli` as we'll need to enforce a special version of Node on the client. I guess it might be easier if we just use a library instead of `Intl` in order to avoid the environment requirement? 🤔

Reply:
Super good callout - we'll look into this.

Reply:
https://nodejs.org/docs/latest-v10.x/api/intl.html#intl_embed_the_entire_icu_full_icu

Reply:
@davidchin with the current state of the helper, this should simply fall back to returning `price.formatted` in the case where the proper `Intl` support is not available. I think that's a pretty graceful fallback.

If we go forward with this, we should probably publish instructions on how to set up node with the full ICU data. As long as this is just a helper, that's probably fine. If we start using this helper in Cornerstone, we'll want to make sure all devs have an appropriate environment.

I found this which looks useful: https://www.npmjs.com/package/full-icu

Reply:
Yeah that could work. We can have that as a dependency for the renderer and dev renderer and specify `NODE_ICU_DATA` environment variable when we start the server.

Just a side note, I think `Intl.NumberFormat` won't throw an error if full ICU data is not installed. It would just return the number formatted using English locale. It would only fallback to the number format configured by the merchant if you provide an invalid locale ID. Anyway, it shouldn't be a problem as long as we can get ICU support working. 😄

Reply:
Gotcha - I was going off of what I saw here: https://nodejs.org/docs/latest-v10.x/api/intl.html#intl_detecting_internationalization_support

I was going to add a similar check:

```
const hasFullICU = (() => {
  try {
    const january = new Date(9e8);
    const spanish = new Intl.DateTimeFormat('es', { month: 'long' });
    return spanish.format(january) === 'enero';
  } catch (err) {
    return false;
  }
})();
```

... but it seemed like my existing try/catch was doing something similar.

I could always add this additional check though!

Reply:
Actually now that I look at this with fresh eyes, I see this check is comparing a static result based on what it *would* be if the Intl libraries were properly installed. I'll roll this in...

Reply:
This is now resolved as we have upgraded node versions such that ICU is included by default.